src/vg/civcraft/mc/namelayer/listeners/MercuryMessageListener.java

```diff
@@ -49,5 +88,29 @@ else if (reason.equals("transfer")){
 			gm.invalidateCache(g.getName());
 		}
 	}
+	
+	@EventHandler(priority = EventPriority.HIGHEST)
+	public void onLogin(PlayerJoinEvent event){
+		MercuryPlugin.handler.sendMessage("all", "name_layer", "login "+event.getPlayer().getDisplayName());
+		Bukkit.getLogger().info("LOGIN");
+	}
+	
+	@EventHandler(priority = EventPriority.HIGHEST)
+	public void onLogoff(PlayerQuitEvent event){
+		MercuryPlugin.handler.sendMessage("all", "name_layer", "logoff "+event.getPlayer().getDisplayName());
+		Bukkit.getLogger().info("LOGOFF");
+	}
 
+	@Override
+	public void run() {
+		NameLayerPlugin nl = NameLayerPlugin.getInstance();
+		nl.isMercuryEnabled = Bukkit.getPluginManager().isPluginEnabled("Mercury");
+		if (nl.isMercuryEnabled){
+			nl.getServer().getPluginManager().registerEvents(this, nl);
```

Comment:
You want it to make another MercuryMessageListener? How come? Wouldn't nl.getServer().getPluginManager().registerEvents(this, nl); increment the reference so it wouldnt get gc'd when the runnable is over?